ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -46,57 +46,59 @@ final class TopicDeletionEndpoint[F[_]: Futurable] (deletionProgram: TopicDeleti
     kafkaResults.flatMap(_.kafkaDeleteTopicErrorList.errors.map(e => DeletionEndpointResponse(e.topicName, e.errorMessage)).toList)
   }
 
-  private def validResponse(topics: List[String], userDeleting: String)(implicit ec: ExecutionContext) = {
+  private def validResponse(topics: List[String], userDeleting: String, path: String)(implicit ec: ExecutionContext) = {
     topics.foreach { topicName =>
-      addPromHttpMetric(topicName, StatusCodes.OK.toString(), "/deleteTopics/")
+      addPromHttpMetric(topicName, StatusCodes.OK.toString(), path)
       log.info(s"User $userDeleting deleted topic: $topicName")
     }
     complete(StatusCodes.OK, topics)
   }
 
-  private def returnResponse(topics: List[String], userDeleting: String, response: List[DeletionEndpointResponse])(implicit ec: ExecutionContext) = {
-    if (response.length == topics.length) {
-      topics.map(topicName => addPromHttpMetric(topicName, StatusCodes.InternalServerError.toString(), "/deleteTopics"))
-      complete(StatusCodes.InternalServerError, response)
-    } else {
+  private def returnResponse(topics: List[String], userDeleting: String, response: List[DeletionEndpointResponse], path: String)(implicit ec: ExecutionContext) = {
+    if(response.map(_.topicOrSubject.contains("-key")).contains(true) ||
```

Comment:
Just FYI, this could also be written as:

```scala
response.exists(r => r.topicOrSubject.contains("-key") || r.topicOrSubject.contains("-value"))
```

`.exists` defaults to false, unless the predicate (the code inside of the exists function) returns true. In that way it is basically a `contains`, but it accepts a function instead of just a parameter to check for.